tests/native_api_tests/test_api_model.cpp

```diff
@@ -857,14 +875,18 @@ TEST_CASE_TEMPLATE(
         } else {
             update_dataset.add_buffer("sym_load", -1, 2, sym_load_indptr.data(), sym_load_update_buffer);
         }
+        update_dataset.add_buffer("source", -1, source_indptr.back(), source_indptr.data(), source_update_buffer);
```

Comment:
Shouldn't this go in the sparse `if` statement from above? I know that when the ID is not there for the source it is a sparse representation of a dense buffer, but it still has an indptr.

Reply:
the `sparsity_type` affects the `sym_load`. I did not want to add more complexity to `source`, especially not because this particular issue solved in this PR is about mixing of multiple types

Reply:
will add a comment